tests/testExposure.py

```diff
@@ -154,6 +154,35 @@ def testGetMaskedImage(self):
         self.assertEqual(crOnlyWidth, self.exposureCrOnly.getWidth())
         self.assertEqual(crOnlyHeight, self.exposureCrOnly.getHeight())
 
+
+    def testProperties(self):
+        self.assertMaskedImagesEqual(self.exposureMiOnly.maskedImage,
+                                     self.exposureMiOnly.getMaskedImage())
+        mi2 = afwImage.MaskedImageF(self.exposureMiOnly.getDimensions())
+        mi2.image.array = 5.0
+        mi2.variance.array = 3.0
+        mi2.mask.array = 0x1
+        self.exposureMiOnly.maskedImage = mi2
+        self.assertMaskedImagesEqual(self.exposureMiOnly.maskedImage, mi2)
+        self.assertImagesEqual(self.exposureMiOnly.image,
+                               self.exposureMiOnly.maskedImage.image)
+
+        image3 = afwImage.ImageF(self.exposureMiOnly.getDimensions())
+        image3.array = 3.0
+        self.exposureMiOnly.image = image3
+        self.assertImagesEqual(self.exposureMiOnly.image, image3)
+
+        mask3 = afwImage.MaskU(self.exposureMiOnly.getDimensions())
+        mask3.array = 0x2
+        self.exposureMiOnly.mask = mask3
+        self.assertMasksEqual(self.exposureMiOnly.mask, mask3)
+
+        var3 = afwImage.ImageF(self.exposureMiOnly.getDimensions())
+        var3.array = 2.0
+        self.exposureMiOnly.variance = var3
+        self.assertImagesEqual(self.exposureMiOnly.variance, var3)
+
+
```

Comment:
I had considered the ability to assign a scalar without extra brackets a feature, not a bug, but it's not something I'm wedded to.  I wonder what other Python style experts think (@timj, @parejkoj, @natelust?).

However, I do think it's hard to get coercion of arrays with other types without supporting scalars; I think it's either both or neither.

Subscripted setting should just delegate to our `__getitem__` and NumPy's `__setitem__`, but I'm happy to test it as well.

It occurs to me that what really needs to be tested is in-place binary operators.  I'm not sure if those will work unless I do implement `__setitem__` via NumPy and hence support assigning scalars, but I'll see.